test/ILLink.RoslynAnalyzer.Tests/LinkerTestCases.cs

```diff
@@ -19,14 +19,10 @@ public void RequiresCapability (MethodDeclarationSyntax m, List<AttributeSyntax>
 		{
 			switch (m.Identifier.ValueText) {
 			case "MethodWithDuplicateRequiresAttribute":
-			case "TestCovariantReturnCallOnDerived":
 			case "TestRequiresUnreferencedCodeOnlyThroughReflection":
-			case "TestStaticCctorRequiresUnreferencedCode":
-			case "TestStaticCtorMarkingIsTriggeredByFieldAccess":
-			case "TestTypeWhichOverridesMethodVirtualMethodRequiresUnreferencedCode":
 			case "TestRequiresInMethodFromCopiedAssembly":
 			case "TestRequiresThroughReflectionInMethodFromCopiedAssembly":
-			case "TestStaticCtorTriggeredByMethodCall":
+			// The way roslyn and linker name the source location is different for BeforeFieldInit
```

Comment:
I think I don't understand this comment

Reply:
There is a discrepancy between the way the linker and roslyn represent the source location for this warning. Roslyn will use (8,29) and the linker will print Mono.Linker.Tests.Cases.RequiresCapability.RequiresUnreferencedCodeCapability.TypeIsBeforeFieldInit..cctor()

**Roslyn warning**
(8,29): warning IL2026: Using method 'C.TypeIsBeforeFieldInit.AnnotatedMethod()' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code. Message from --TypeIsBeforeFieldInit.AnnotatedMethod--.

**Linker warning**
IL2026: Mono.Linker.Tests.Cases.RequiresCapability.RequiresUnreferencedCodeCapability.TypeIsBeforeFieldInit..cctor(): Using method 'Mono.Linker.Tests.Cases.RequiresCapability.RequiresUnreferencedCodeCapability.TypeIsBeforeFieldInit.AnnotatedMethod()' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code. Message from --TypeIsBeforeFieldInit.AnnotatedMethod--.

Since linker is using LogContains the Roslyn test will try to match the `TypeIsBeforeFieldInit..cctor()` value against his diagnostic value and fails. The scenario is supported, therefore the addition of the test in the RequiresUnreferencedCodeAnalyzerTests.cs file. But you are right because I try to keep the comment short might not be understandable. I will update it.